lib/pages/home_page.dart

```diff
@@ -16,11 +20,43 @@ class HomePage extends StatefulWidget {
 
 class _HomePageState extends State<HomePage> {
   Position? pos;
+  Marker daily = Marker(point: LatLng(0, 0), builder: (context) => Container());
+  int distance =
+      DistCalculator.instance.getDist(Location(0, 0), Location(0, 0));
+
+  Timer? _timer;
 
   Future<void> getPosition() async {
     pos = await Geolocation.instance.position;
   }
 
+  Future<void> initiateChallenge() async {
+    setState(() {
+      daily = DistCalculator.instance
+          .initiateChallenge(2500, 500000, LatLng(pos!.latitude, pos!.longitude));
+      distance = DistCalculator.instance.getDist(
+          Location(pos!.latitude, pos!.longitude),
+          Location(daily.point.latitude, daily.point.longitude));
+      _timer = Timer.periodic(const Duration(seconds: 10), (_) async {
+        Position pos = await Geolocation.instance.position;
+        if (DistCalculator.instance.checkDist(
+            Location(pos.latitude, pos.longitude),
+            Location(daily.point.latitude, daily.point.longitude))) {
+          finishChallenge();
+        }
+      });
+    });
+  }
+
+  Future<void> finishChallenge() async {
+    _timer!.cancel();
```

Comment:
Budeš mít bezpečnější udělat `_timer?.cancel()`. Takže se cancel zavolá pouze pokud je nad čím. Takhle kdybys někde změnil třeba logiku tak že to může být tady null tak ti spadne appka 😉 (a celkově pokud to okolnosti dovolují je lepší používat **?**, je to mnohem víc safe)